tests/WebUI/ActionsCreatorTests/ImageDetailsActionsCreator.test.ts

```diff
@@ -0,0 +1,56 @@
+import { ImageDetailsActionsCreator } from "../../../src/WebUI/ImageDetails/ImageDetailsActionsCreator";
+import { ImageDetailsStore } from "../../../src/WebUI/ImageDetails/ImageDetailsStore";
+import { StoreManager } from "../../../src/WebUI/FluxCommon/StoreManager";
+import { ActionsCreatorManager } from "../../../src/WebUI/FluxCommon/ActionsCreatorManager";
+import { IImageDetails } from "../../../src/Contracts/Types";
+import { KubeFactory } from "../../../src/WebUI/KubeFactory";
+import { MockImageService } from "../MockImageService";
+
+describe("ImageDetailsActionsCreator getImageDetails Tests", () => {
+    const mockImageDetails: IImageDetails = {
```

Comment:
let us use data like the following, we can remove unnecessary data from this

{
            "imageName": "https://docker.io/aksdockersample/aks-docker-sample@sha256:8c8a34af9ed6bc3f50b019fc35fa43f692e7933203072db2281f36de54dc3527",
            "imageUri": "https://docker.io/aksdockersample/aks-docker-sample@sha256:8c8a34af9ed6bc3f50b019fc35fa43f692e7933203072db2281f36de54dc3527",
            "hash": "8c8a34af9ed6bc3f50b019fc35fa43f692e7933203072db2281f36de54dc3527",
            "distance": null,
            "imageType": "",
            "mediaType": "",
            "tags": [
                "9388636",
                "latest"
            ],
            "layerInfo": [
                {
                    "directive": "ADD",
                    "arguments": "file:3e6141c0c9cb74b14a281eb3ab7aaf162a625733e652c3948b323bb2ec8b4343 in /",
                    "size": "127MB",
                    "createdOn": "/Date(1524897855000)/"
                },
                {
                    "directive": "CMD",
                    "arguments": "[\"bash\"]",
                    "size": "0B",
                    "createdOn": "/Date(1524897856000)/"
                },
                {
                    "directive": "RUN",
                    "arguments": "/bin/sh -c apt-get update && apt-get install -y --no-install-recommends   ca-certificates   curl   netbase   wget  && rm -rf /var/lib/apt/lists/",
                    "size": "41.2MB",
                    "createdOn": "/Date(1528240063000)/"
                },
                {
                    "directive": "RUN",
                    "arguments": "/bin/sh -c set -ex;  if ! command -v gpg > /dev/null; then   apt-get update;   apt-get install -y --no-install-recommends    gnupg    dirmngr   ;   rm -rf /var/lib/apt/lists/*;  f",
                    "size": "0B",
                    "createdOn": "/Date(1528240064000)/"
                },
                {
                    "directive": "RUN",
                    "arguments": "/bin/sh -c apt-get update && apt-get install -y --no-install-recommends   bzr   git   mercurial   openssh-client   subversion     procps  && rm -rf /var/lib/apt/lists/",
                    "size": "124MB",
                    "createdOn": "/Date(1528240128000)/"
                },
                {
                    "directive": "RUN",
                    "arguments": "/bin/sh -c set -ex;  apt-get update;  apt-get install -y --no-install-recommends   autoconf   automake   bzip2   dpkg-dev   file   g++   gcc   imagemagick   libbz2-dev   libc6-dev   libcurl4-openssl-dev   libdb-dev   libevent-dev   libffi-dev   libgdbm-dev   libgeoip-dev   libglib2.0-dev   libjpeg-dev   libkrb5-dev   liblzma-dev   libmagickcore-dev   libmagickwand-dev   libncurses5-dev   libncursesw5-dev   libpng-dev   libpq-dev   libreadline-dev   libsqlite3-dev   libssl-dev   libtool   libwebp-dev   libxml2-dev   libxslt-dev   libyaml-dev   make   patch   xz-utils   zlib1g-dev     $(    if apt-cache show 'default-libmysqlclient-dev' 2>/dev/null | grep -q '^Version:'; then     echo 'default-libmysqlclient-dev';    else     echo 'libmysqlclient-dev';    fi   )  ;  rm -rf /var/lib/apt/lists/",
                    "size": "320MB",
                    "createdOn": "/Date(1528240243000)/"
                },
                {
                    "directive": "RUN",
                    "arguments": "/bin/sh -c groupadd --gid 1000 node   && useradd --uid 1000 --gid node --shell /bin/bash --create-home nod",
                    "size": "335kB",
                    "createdOn": "/Date(1528316012000)/"
                },
                {
                    "directive": "RUN",
                    "arguments": "/bin/sh -c set -ex   && for key in     94AE36675C464D64BAFA68DD7434390BDBE9B9C5     FD3A5288F042B6850C66B31F09FE44734EB7990E     71DCFD284A79C3B38668286BC97EC7A07EDE3FC1     DD8F2338BAE7501E3DD5AC78C273792F7D83545D     C4F0DFFF4E8C1A8236409D08E73BC641CC11F4C8     B9AE9905FFD7803F25714661B63B535A4C206CA9     56730D5401028683275BD23C23EFEFE93C4CFFFE     77984A986EBC2AA786BC0F66B01FBB92821C587A   ; do     gpg --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys \"$key\" ||     gpg --keyserver hkp://ipv4.pool.sks-keyservers.net --recv-keys \"$key\" ||     gpg --keyserver hkp://pgp.mit.edu:80 --recv-keys \"$key\" ;   don",
                    "size": "129kB",
                    "createdOn": "/Date(1528316014000)/"
                },
                {
                    "directive": "ENV",
                    "arguments": "NODE_VERSION=6.14.2",
                    "size": "0B",
                    "createdOn": "/Date(1528316117000)/"
                },
                {
                    "directive": "RUN",
                    "arguments": "/bin/sh -c ARCH= && dpkgArch=\"$(dpkg --print-architecture)\"   && case \"${dpkgArch##*-}\" in     amd64) ARCH='x64';;     ppc64el) ARCH='ppc64le';;     s390x) ARCH='s390x';;     arm64) ARCH='arm64';;     armhf) ARCH='armv7l';;     i386) ARCH='x86';;     *) echo \"unsupported architecture\"; exit 1 ;;   esac   && curl -fsSLO --compressed \"https://nodejs.org/dist/v$NODE_VERSION/node-v$NODE_VERSION-linux-$ARCH.tar.xz\"   && curl -fsSLO --compressed \"https://nodejs.org/dist/v$NODE_VERSION/SHASUMS256.txt.asc\"   && gpg --batch --decrypt --output SHASUMS256.txt SHASUMS256.txt.asc   && grep \" node-v$NODE_VERSION-linux-$ARCH.tar.xz\\$\" SHASUMS256.txt | sha256sum -c -   && tar -xJf \"node-v$NODE_VERSION-linux-$ARCH.tar.xz\" -C /usr/local --strip-components=1 --no-same-owner   && rm \"node-v$NODE_VERSION-linux-$ARCH.tar.xz\" SHASUMS256.txt.asc SHASUMS256.txt   && ln -s /usr/local/bin/node /usr/local/bin/nodej",
                    "size": "43.1MB",
                    "createdOn": "/Date(1528316121000)/"
                },
                {
                    "directive": "ENV",
                    "arguments": "YARN_VERSION=1.6.0",
                    "size": "0B",
                    "createdOn": "/Date(1528316121000)/"
                },
                {
                    "directive": "RUN",
                    "arguments": "/bin/sh -c set -ex   && for key in     6A010C5166006599AA17F08146C2130DFD2497F5   ; do     gpg --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys \"$key\" ||     gpg --keyserver hkp://ipv4.pool.sks-keyservers.net --recv-keys \"$key\" ||     gpg --keyserver hkp://pgp.mit.edu:80 --recv-keys \"$key\" ;   done   && curl -fsSLO --compressed \"https://yarnpkg.com/downloads/$YARN_VERSION/yarn-v$YARN_VERSION.tar.gz\"   && curl -fsSLO --compressed \"https://yarnpkg.com/downloads/$YARN_VERSION/yarn-v$YARN_VERSION.tar.gz.asc\"   && gpg --batch --verify yarn-v$YARN_VERSION.tar.gz.asc yarn-v$YARN_VERSION.tar.gz   && mkdir -p /opt   && tar -xzf yarn-v$YARN_VERSION.tar.gz -C /opt/   && ln -s /opt/yarn-v$YARN_VERSION/bin/yarn /usr/local/bin/yarn   && ln -s /opt/yarn-v$YARN_VERSION/bin/yarnpkg /usr/local/bin/yarnpkg   && rm yarn-v$YARN_VERSION.tar.gz.asc yarn-v$YARN_VERSION.tar.g",
                    "size": "4.47MB",
                    "createdOn": "/Date(1528316124000)/"
                },
                {
                    "directive": "CMD",
                    "arguments": "[\"node\"]",
                    "size": "0B",
                    "createdOn": "/Date(1528316124000)/"
                },
                {
                    "directive": "EXPOSE",
                    "arguments": "8181",
                    "size": "0B",
                    "createdOn": "/Date(1555512651000)/"
                },
                {
                    "directive": "COPY",
                    "arguments": "file:c535f3db468b26bc8607d4b28f2b21e30330cd9a221f32808a00eeb3eb93908c in .",
                    "size": "567B",
                    "createdOn": "/Date(1555512653000)/"
                },
                {
                    "directive": "COPY",
                    "arguments": "file:f2255bedc2cb1900a54715e2587b984d579c4d49a4ea4ed5a63d53b9d1f4f7dd in .",
                    "size": "583B",
                    "createdOn": "/Date(1555512654000)/"
                },
                {
                    "directive": "CMD",
                    "arguments": "[\"/bin/sh\" \"-c\" \"node server.js\"]",
                    "size": "0B",
                    "createdOn": "/Date(1555512655000)/"
                },
                {
                    "directive": "LABEL",
                    "arguments": "com.azure.dev.image.build.buildnumber=20190417.4",
                    "size": "0B",
                    "createdOn": "/Date(1555512656000)/"
                },
                {
                    "directive": "LABEL",
                    "arguments": "com.azure.dev.image.build.builduri=vstfs:///Build/Build/9275537",
                    "size": "0B",
                    "createdOn": "/Date(1555512658000)/"
                },
                {
                    "directive": "LABEL",
                    "arguments": "com.azure.dev.image.build.definitionname=madhuv-msft.DockerAKSTutorial",
                    "size": "0B",
                    "createdOn": "/Date(1555512659000)/"
                },
                {
                    "directive": "LABEL",
                    "arguments": "com.azure.dev.image.build.repository.name=madhuv-msft/DockerAKSTutorial",
                    "size": "0B",
                    "createdOn": "/Date(1555512660000)/"
                },
                {
                    "directive": "LABEL",
                    "arguments": "com.azure.dev.image.build.repository.uri=https://github.com/madhuv-msft/DockerAKSTutorial",
                    "size": "0B",
                    "createdOn": "/Date(1555512662000)/"
                },
                {
                    "directive": "LABEL",
                    "arguments": "com.azure.dev.image.build.requestedfor=Madhu V",
                    "size": "0B",
                    "createdOn": "/Date(1555512663000)/"
                },
                {
                    "directive": "LABEL",
                    "arguments": "com.azure.dev.image.build.sourcebranchname=master",
                    "size": "0B",
                    "createdOn": "/Date(1555512664000)/"
                },
                {
                    "directive": "LABEL",
                    "arguments": "com.azure.dev.image.build.sourceversion=f62a0839bcd9fe1393407aa74d68064e207bd749",
                    "size": "0B",
                    "createdOn": "/Date(1555512666000)/"
                },
                {
                    "directive": "LABEL",
                    "arguments": "com.azure.dev.image.system.teamfoundationcollectionuri=https://dev.azure.com/mseng/",
                    "size": "0B",
                    "createdOn": "/Date(1555512667000)/"
                },
                {
                    "directive": "LABEL",
                    "arguments": "com.azure.dev.image.system.teamproject=AzureDevOps",
                    "size": "0B",
                    "createdOn": "/Date(1555512668000)/"
                },
                {
                    "directive": "EXPOSE",
                    "arguments": "8181",
                    "size": "0B",
                    "createdOn": "/Date(1556796604000)/"
                },
                {
                    "directive": "COPY",
                    "arguments": "file:c535f3db468b26bc8607d4b28f2b21e30330cd9a221f32808a00eeb3eb93908c in .",
                    "size": "567B",
                    "createdOn": "/Date(1556796605000)/"
                },
                {
                    "directive": "COPY",
                    "arguments": "file:f2255bedc2cb1900a54715e2587b984d579c4d49a4ea4ed5a63d53b9d1f4f7dd in .",
                    "size": "583B",
                    "createdOn": "/Date(1556796606000)/"
                },
                {
                    "directive": "CMD",
                    "arguments": "[\"/bin/sh\" \"-c\" \"node server.js\"]",
                    "size": "0B",
                    "createdOn": "/Date(1556796607000)/"
                },
                {
                    "directive": "LABEL",
                    "arguments": "com.azure.dev.image.build.buildnumber=20190502.1",
                    "size": "0B",
                    "createdOn": "/Date(1556796608000)/"
                },
                {
                    "directive": "LABEL",
                    "arguments": "com.azure.dev.image.build.builduri=vstfs:///Build/Build/9388636",
                    "size": "0B",
                    "createdOn": "/Date(1556796610000)/"
                },
                {
                    "directive": "LABEL",
                    "arguments": "com.azure.dev.image.build.definitionname=madhuv-msft.DockerAKSTutorial",
                    "size": "0B",
                    "createdOn": "/Date(1556796611000)/"
                },
                {
                    "directive": "LABEL",
                    "arguments": "com.azure.dev.image.build.repository.name=madhuv-msft/DockerAKSTutorial",
                    "size": "0B",
                    "createdOn": "/Date(1556796612000)/"
                },
                {
                    "directive": "LABEL",
                    "arguments": "com.azure.dev.image.build.repository.uri=https://github.com/madhuv-msft/DockerAKSTutorial",
                    "size": "0B",
                    "createdOn": "/Date(1556796614000)/"
                },
                {
                    "directive": "LABEL",
                    "arguments": "com.azure.dev.image.build.requestedfor=Madhu V",
                    "size": "0B",
                    "createdOn": "/Date(1556796615000)/"
                },
                {
                    "directive": "LABEL",
                    "arguments": "com.azure.dev.image.build.sourcebranchname=master",
                    "size": "0B",
                    "createdOn": "/Date(1556796616000)/"
                },
                {
                    "directive": "LABEL",
                    "arguments": "com.azure.dev.image.build.sourceversion=c3c5de25e921d7f92ad83e85f08ce354ced86329",
                    "size": "0B",
                    "createdOn": "/Date(1556796618000)/"
                },
                {
                    "directive": "LABEL",
                    "arguments": "com.azure.dev.image.system.teamfoundationcollectionuri=https://dev.azure.com/mseng/",
                    "size": "0B",
                    "createdOn": "/Date(1556796619000)/"
                },
                {
                    "directive": "LABEL",
                    "arguments": "com.azure.dev.image.system.teamproject=AzureDevOps",
                    "size": "0B",
                    "createdOn": "/Date(1556796620000)/"
                }
            ],
            "runId": 9388636,
            "pipelineVersion": "20190502.1",
            "pipelineName": "madhuv-msft.DockerAKSTutorial",
            "pipelineId": "8462",
            "imageSize": "2836",
            "jobName": "Build job",
            "createTime": "/Date(1556796627018)/"
        }